### PR TITLE
Adds perfect-numbers exercise

### PR DESCRIPTION
**Pull request by amscotti**
Hi, this PR is to add a new exercise for `perfect-numbers`, [here is a link to the problem-specifications](https://github.com/exercism/problem-specifications/tree/master/exercises/perfect-numbers)

I'm still learning Dart, so feel free to point out anything that needs to be change or improved. I'm also happy to change the `config.json` to better fit your layout for the track.

Thanks! :)

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -381,6 +381,20 @@
         "sorting",
         "trees"
       ]
+    },
+    {
+      "slug": "perfect-numbers",
+      "uuid": "db9eb481-a2b5-4b53-9825-a20797f930de",
+      "core": false,
+      "unlocked_by": null,
+      "difficulty": 3,
```
Comment (Stargator) [inline config.json]: I recommend a difficulty of 2, but I am open to leaving it as 3.

File: config.json [hunk 2]
```diff
@@ -381,6 +381,20 @@
         "sorting",
         "trees"
       ]
+    },
+    {
+      "slug": "perfect-numbers",
+      "uuid": "db9eb481-a2b5-4b53-9825-a20797f930de",
+      "core": false,
+      "unlocked_by": null,
```
Comment (Stargator) [inline config.json]: I suggest moving it so it is unlocked by armstrong-numbers

File: exercises/perfect-numbers/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+- **Perfect**: aliquot sum = number
+  - 6 is a perfect number because (1 + 2 + 3) = 6
+  - 28 is a perfect number because (1 + 2 + 4 + 7 + 14) = 28
+- **Abundant**: aliquot sum > number
+  - 12 is an abundant number because (1 + 2 + 3 + 4 + 6) = 16
+  - 24 is an abundant number because (1 + 2 + 3 + 4 + 6 + 8 + 12) = 36
+- **Deficient**: aliquot sum < number
+  - 8 is a deficient number because (1 + 2 + 4) = 7
+  - Prime numbers are deficient
```
[the remark this answers is not in the document]
Comment (amscotti) [reply exercises/perfect-numbers/README.md]: This doesn't seem only related to the Dart track, so wouldn't an update like this be better made in `problem-specifications`? Then all tracks would have the benefit of having a clearer description.

File: exercises/perfect-numbers/lib/example.dart [hunk 4]
```diff
@@ -0,0 +1,22 @@
+class PerfectNumbers {
+  String classify(int num) {
```
[the remark this answers is not in the document]
Comment (amscotti) [reply exercises/perfect-numbers/lib/example.dart]: Ah, good call! I missed that.

File: exercises/perfect-numbers/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Stargator) [reply exercises/perfect-numbers/README.md]: Yeah, I agree. But for now, we can keep it just in the Dart track. I'm focused on other work at this time. So if you would like to submit a PR there for it, you can.

File: config.json (hunk 2 above)
Comment (Stargator) [reply config.json]: @amscotti This can be reverted. It can be unlocked by difference-of-squares

Comment (amscotti) [reply config.json]: @Stargator Update

File: exercises/perfect-numbers/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Stargator) [reply exercises/perfect-numbers/README.md]: @amscotti This still needs to be addressed

**Pull request closed without merging**